packages/dds/tree/src/tableSchema.ts

```diff
@@ -398,8 +398,16 @@ export namespace System_TableSchema {
 			): void {
 				// TODO: throw if column does not exist in the owning table.
 
-				const columnId = typeof columnOrId === "string" ? columnOrId : columnOrId.id;
-				this.cells.set(columnId, value);
+				if (value === undefined) {
+					this.removeCell(columnOrId);
+				} else {
+					const columnId = typeof columnOrId === "string" ? columnOrId : columnOrId.id;
+
+					// Record nodes' typing does not support setting properties with their insertable types,
+					// but doing so is supported at runtime.
+					// So we will cast the insertable type to the expected value type.
+					this.cells[columnId] = value as CellValueType;
```

Comment:
When facing cases like this (which happens to customers a lot) I think the recommendation should be something like:
```suggestion
					this.cells[columnId] = TreeAlpha.create(CellValueTypeSchema, value);
```

Reply:
This required some additional casting that could use a sanity check, but neither was particularly surprising to me.